src/components/TopicPage.jsx

```diff
@@ -4,25 +4,42 @@ import { useEffect, useState } from "react"
 import Loading from "./Loading"
 import ErrorHandling from "./ErrorHandling"
 import ArticleCards from "./ArticleCards"
+import { useSearchParams } from 'react-router-dom'
+import SortBar from "./SortBar"
+import PageNav from "./PageNav"
 
-function TopicPage() {
+
+function TopicPage({articlesToBrowse, setArticlesToBrowse, sortBy, setSortBy, order, setOrder, limit, setLimit, page, setPage}) {
     const {topic_slug} = useParams()
     const [isLoading, setIsLoading] = useState(true)
     const [error, setError] = useState(null)
-    const [articlesByTopic, setArticlesByTopic] = useState([])
+    let [searchParams, setSearchParams] = useSearchParams()
+
+    let queries = [...searchParams]
```

Comment:
This certainly isn't the best way to do this, you should just access the values you expect from search params and have default values should the not exist

Reply:
also, this should be a `const`